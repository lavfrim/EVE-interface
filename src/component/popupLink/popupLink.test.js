import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import PopupLink from './index';

const corporation = {
    name: 'corp name',
}
describe('PopupLink should render correctly', () => {
    it('with isOpen = false', () => {
        const wrapper = mount(
            <PopupLink corporation={corporation} />
        );
        wrapper.setState({  isOpen: false, stage: 0 });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with stage = basic', () => {
        const wrapper = mount(
            <PopupLink corporation={corporation} />
        );
        wrapper.setState({  isOpen: true, stage: 0 });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with stage = clicked', () => {
        const wrapper = mount(
            <PopupLink corporation={corporation} />
        );
        wrapper.setState({  isOpen: true, stage: 1 });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with stage = returned', () => {
        const wrapper = mount(
            <PopupLink corporation={corporation} />
        );
        wrapper.setState({ isOpen: true, stage: -1 });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
