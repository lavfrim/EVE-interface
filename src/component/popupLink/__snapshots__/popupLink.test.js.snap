// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupLink should render correctly with isOpen = false 1`] = `
<PopupLink
  corporation={
    Object {
      "name": "corp name",
    }
  }
>
  <div
    className="pop-up-link"
    onClick={[Function]}
  >
    corp name
  </div>
</PopupLink>
`;

exports[`PopupLink should render correctly with stage = basic 1`] = `
<PopupLink
  corporation={
    Object {
      "name": "corp name",
    }
  }
>
  <div
    className="pop-up-link"
    onClick={[Function]}
  >
    corp name
  </div>
  <PopUp
    handleClickBackward={[Function]}
    handleClickForward={[Function]}
    handleClose={[Function]}
    stage={0}
  >
    <div
      className="pop-up"
      id="outside"
      onClick={[Function]}
    >
      <div
        className="pop-up__content-container"
      >
        <button
          className="pop-up__content-container__close-button"
          id="close"
        >
          X
        </button>
        <div
          className="pop-up__content-container__content"
        >
          <PopupCard
            corporation={
              Object {
                "name": "corp name",
              }
            }
            handleClickCEO={[Function]}
            stage={0}
          >
            <CorporationCard
              ceoInfo={Object {}}
              corporation={
                Object {
                  "name": "corp name",
                }
              }
              handleClickCEO={[Function]}
            >
              <p>
                Corporation name: corp name
              </p>
              <div
                className="corporation-card__ceo-link"
              >
                <p
                  className="corporation-card__ceo-link-description"
                >
                  CEO name:
                </p>
                <Loading>
                  <span>
                    Loading...
                  </span>
                </Loading>
              </div>
              <p>
                Member count: undefined
              </p>
              <p>
                Description: undefined
              </p>
            </CorporationCard>
          </PopupCard>
        </div>
      </div>
    </div>
  </PopUp>
</PopupLink>
`;

exports[`PopupLink should render correctly with stage = clicked 1`] = `
<PopupLink
  corporation={
    Object {
      "name": "corp name",
    }
  }
>
  <div
    className="pop-up-link"
    onClick={[Function]}
  >
    corp name
  </div>
  <PopUp
    handleClickBackward={[Function]}
    handleClickForward={[Function]}
    handleClose={[Function]}
    stage={1}
  >
    <div
      className="pop-up"
      id="outside"
      onClick={[Function]}
    >
      <div
        className="pop-up__content-container"
      >
        <button
          className="pop-up__content-container__close-button"
          id="close"
        >
          X
        </button>
        <div
          className="pop-up__content-container__content"
        >
          <PopupCard
            corporation={
              Object {
                "name": "corp name",
              }
            }
            handleClickCEO={[Function]}
            stage={1}
          >
            <CEOCard
              ceoInfo={Object {}}
            >
              <p>
                CEO name: undefined
              </p>
              <p>
                Birthday: Invalid Date
              </p>
              <div
                className="ceo-card__race"
              >
                <p
                  className="ceo-card__race-description"
                >
                  Race:
                </p>
                <Loading>
                  <span>
                    Loading...
                  </span>
                </Loading>
              </div>
            </CEOCard>
          </PopupCard>
        </div>
        <button
          className="pop-up__content-container__nav-button"
          id="backward"
        >
          &lt;
        </button>
      </div>
    </div>
  </PopUp>
</PopupLink>
`;

exports[`PopupLink should render correctly with stage = returned 1`] = `
<PopupLink
  corporation={
    Object {
      "name": "corp name",
    }
  }
>
  <div
    className="pop-up-link"
    onClick={[Function]}
  >
    corp name
  </div>
  <PopUp
    handleClickBackward={[Function]}
    handleClickForward={[Function]}
    handleClose={[Function]}
    stage={-1}
  >
    <div
      className="pop-up"
      id="outside"
      onClick={[Function]}
    >
      <div
        className="pop-up__content-container"
      >
        <button
          className="pop-up__content-container__close-button"
          id="close"
        >
          X
        </button>
        <div
          className="pop-up__content-container__content"
        >
          <PopupCard
            corporation={
              Object {
                "name": "corp name",
              }
            }
            handleClickCEO={[Function]}
            stage={-1}
          >
            <CorporationCard
              ceoInfo={Object {}}
              corporation={
                Object {
                  "name": "corp name",
                }
              }
              handleClickCEO={[Function]}
            >
              <p>
                Corporation name: corp name
              </p>
              <div
                className="corporation-card__ceo-link"
              >
                <p
                  className="corporation-card__ceo-link-description"
                >
                  CEO name:
                </p>
                <Loading>
                  <span>
                    Loading...
                  </span>
                </Loading>
              </div>
              <p>
                Member count: undefined
              </p>
              <p>
                Description: undefined
              </p>
            </CorporationCard>
          </PopupCard>
        </div>
        <button
          className="pop-up__content-container__nav-button"
          id="forward"
        >
          &gt;
        </button>
      </div>
    </div>
  </PopUp>
</PopupLink>
`;
